chimerapy/states.py

```diff
@@ -0,0 +1,34 @@
+from typing import List, Dict
+from dataclasses import dataclass, field
+from dataclasses_json import dataclass_json
+
+
+@dataclass_json
+@dataclass
+class NodeState:
+    id: str
+    name: str = ""
+    init: bool = False
+    connected: bool = False
+    ready: bool = False
+    finished: bool = False
```

Comment:
Are these states mutually exclusive? If so, it might be nice to enumerate them.

Reply:
Not necessarily exclusive but it's possible to be `ready` but not `connected`. Therefore, using an `Enum` might not work here.